yumex/service/yumex_updater_systray.py

```diff
@@ -0,0 +1,145 @@
+else:
+    from yumex.service.dnf4 import UpdateChecker
+
+from typing import List, Set
+from packaging import version
+
+# Define paths
+home_dir = Path.home()
+config_dir = home_dir / '.config' / 'yumex'
+default_config_path = '/usr/share/yumex/yumex-service.conf'
+user_config_path = config_dir / 'yumex-service.conf'
+
+# Create the config directory if it doesn't exist
+config_dir.mkdir(parents=True, exist_ok=True)
+
+# Copy the default config file if the user config file doesn't exist
+if not user_config_path.exists():
+    shutil.copy(default_config_path, user_config_path)
+
+# Read configuration file
+config = configparser.ConfigParser()
+config.read(user_config_path)
+
+custom_updater = config.get('DEFAULT', 'custom_updater', fallback=None)
+always_hide = config.getboolean('DEFAULT', 'always_hide', fallback=False)
+update_sync_interval = config.getint('DEFAULT', 'update_sync_interval', fallback=3600)
+
+class UpdateService(dbus.service.Object):
+    def __init__(self, bus_name: dbus.service.BusName, object_path: str) -> None:
+        super().__init__(bus_name, object_path)
+
+    @dbus.service.method('com.yumex.UpdateService')
+    def RefreshUpdates(self) -> None:
+        refresh_updates()
+
+def on_clicked_update(widget: Gtk.Widget) -> None:
+    if custom_updater:
+        subprocess.Popen([custom_updater])
+
+def on_clicked_pm(widget: Gtk.Widget) -> None:
+    subprocess.Popen(["/usr/bin/yumex"])
+
+
+def refresh_updates(widget: Gtk.Widget = None) -> None:
+
+    sys_update_count = len(UpdateChecker())
+
+    flatpak_user_count = len(Flatpak.Installation.new_user().list_installed_refs_for_update())
+    flatpak_sys_count = len(Flatpak.Installation.new_system().list_installed_refs_for_update())
+
+    update_count = sys_update_count + flatpak_user_count + flatpak_sys_count
+
+    hover_text_lines = ["There are updates available:"]
+    if sys_update_count > 0:
+        hover_text_lines.append(f"  System: {sys_update_count}")
+    if flatpak_user_count > 0:
+        hover_text_lines.append(f"  Flatpak (user): {flatpak_user_count}")
+    if flatpak_sys_count > 0:
+        hover_text_lines.append(f"  Flatpak (system): {flatpak_sys_count}")
+    hover_text = "\n".join(hover_text_lines)
+
+    if not always_hide:
+        if update_count > 0:
+            GLib.idle_add(indicator.set_title, hover_text)
+            GLib.idle_add(indicator.set_status, AppIndicator3.IndicatorStatus.ACTIVE)
+        else:
+            GLib.idle_add(indicator.set_status, AppIndicator3.IndicatorStatus.PASSIVE)
+
+def check_updates() -> None:
+    while True:
+        refresh_updates()
+        time.sleep(update_sync_interval)
+
+
+svg_path = Path("/usr/share/icons/hicolor/scalable/apps/yumex-system-software-update.svg")
+png_data = cairosvg.svg2png(url=str(svg_path))
+
+image = Image.open(io.BytesIO(png_data))
+cache_dir = os.path.expanduser("~/.cache")
+image_path = os.path.join(cache_dir, "yumex-system-software-update.png")
+image.save(image_path)
```

Comment:
Why converting the SVG icon to a PNG every time, why not do this at installation time
It will remove depencies cairosvg, PIL

Reply:
It looks like AppIndicator can use a .svg directly

https://stackoverflow.com/questions/49474199/appindicator3-change-icon